contrib/intel/jenkins/tests.py

```diff
@@ -688,58 +693,100 @@ def execute_cmd(self):
 class MpichTestSuite(Test):
 
     def __init__(self, jobname, buildno, testname, core_prov, fabric,
-                 hosts, mpitype, ofi_build_mode, user_env, log_file, util_prov=None):
+                 hosts, mpitype, ofi_build_mode, user_env, log_file, util_prov=None, weekly=None):
 
         super().__init__(jobname, buildno, testname, core_prov,
                          fabric, hosts, ofi_build_mode, user_env, log_file, mpitype,
                          util_prov)
-
-        self.mpichsuitepath = f'{self.middlewares_path}/{mpitype}/'\
-                              'mpichsuite/test/mpi/'
+        self.mpichpath = f'{cloudbees_config.build_dir}/mpich_temp/mpich-4.1/'
+        self.mpichsuitepath = f'{self.mpichpath}/test/mpi/'
+        self.impi_mpichtestpath = f'{cloudbees_config.build_dir}/impi_mpichtest/mpich-test'
         self.pwd = os.getcwd()
         self.mpi_type = mpitype
-
-    def testgroup(self, testgroupname):
-        testpath = f'{self.mpichsuitepath}/{testgroupname}'
-        tests = []
-        with open(f'{testpath}/testlist') as file:
-            for line in file:
-                if(line[0] != '#' and  line[0] != '\n'):
-                    tests.append((line.rstrip('\n')).split(' '))
-
-        return tests
-
-    def set_options(self, nprocs, timeout=None):
-        self.mpi.n = nprocs
-        if (timeout != None):
-            os.environ['MPIEXEC_TIMEOUT']=timeout
-
+        self.weekly = weekly
+
+    def create_hostfile(self, file, hostlist):
+        with open(file, "w") as f:
+            for host in hostlist:
+                f.write(f"{host}\n")
+
+    def update_testlists(self, filename, category):
+        with open(filename, 'r') as file:
+            lines = file.read().splitlines()
+        for line in lines:
+            if (line == category):
+                print("Excluding Tests:" + line)
+                lines[lines.index(line)] = f'#{line}'
+            else:
+                continue
+        with open(filename, 'w') as file:
+            file.write('\n'.join(lines))
+
+    def  exclude_tests(self, test_root, provider):
+        categories = cloudbees_config.mpichtests_exclude[f'{provider}']
```

Comment:
is there a reason the excluded tests are hidden in the config file? The ones for osu were okay to be in an exclude dictionary in the osu class.

Reply:
The thought behind keeping it in a common location was that its accessible from other files in case we need to write code to extract the excluded list of tests in the summary. That also makes it easier to add/remove excludes in the future without having to push code changes and run another PR.

Reply:
you can put the data structure into common.py so that the summary.py file can also access it

Reply:
@nikhilnanal did we come to a decision on this?